### ISSUE TITLE
Dataset generation details for SALM-Duplex

### ISSUE THREAD

**Issue by siddharth1712** [NVIDIA/NeMo#15039]
Hi,

Can you please specify more details on how the dataset for SALM-Duplex model training prepared? The paper mentions Spoken QA dataset generated from Text QA datasets like MSMARCO and Alpaca. 

The paper then mentions that ASR datasets are used as context and QA pairs are generated from snippets of real user audio here: 
"To mitigate overfitting to synthetic data, we follow [29] to create additional synthetic QA pairs using the Mixtral-8x22B LLM from English ASR-labeled data (8k public 2 and 12k in-house). This data is then synthesized using the same TTS, denoted as ASR-QA. The resultant user speech contains both TTS and real data."

How is the actual ASR audio snippet and the TTS generated question combined in the user stream? Are they simply appended one after the other? Is there any added silence duration added between them. How many QA pairs are generated for each sample from the ASR datasets?

Please help or correct me if I am not understanding the dataset generation process specifically for ASR data properly

**Comment by stevehuang52** [comment]
@zhehuaichen could you please help answering this question?

**Comment by zhehuaichen** [comment]
hi thanks for the interest! There are 2 types, a) Spoken QA dataset generated from Text QA datasets  b) ASR-QA
a) is basically a very typical way to create synthetic data, refer to [29] or let us know if any followups
b) `Are they simply appended one after the other?` yes ; `Is there any added silence duration added between them` yes and in the paper it is 0.32 sec. `How many QA pairs are generated for each sample from the ASR datasets` one for each and the resultant amount are in Table 1 of https://arxiv.org/pdf/2505.15670

To help understanding, we will consider releasing the data processing script with the checkpoint release later

Close this thread for now but feel free to reopen or reply with your followups

**Comment by siddharth1712** [comment]
Thank you for your reply and the information related to the ASR-QA dataset, it is very helpful. I have a few followups listed below

1. Are the context TTS voice and question TTS voice kept the same?
2. Can you please share how the InternalSFT dataset was created? What sources/datasets were used? And does this and the Topic and UltraChat datasets have the context part of the conversation in the beginning or are these purely conversational in nature without context.
3. Do you train on single-turn Spoken QA data first and then on multi-turn? Or is the complete training done on 4-turn augmented and real multi-turn data directly? Is there any stages in the training?

**Comment by zhehuaichen** [comment]
1. No. This stems from we don't prompt voices from ASR training data in the context audio
2. These data are designed to simulate conversations in nature. While we may not be able to share exactly InternalSFT, it is similar to https://huggingface.co/datasets/nvidia/Nemotron-Post-Training-Dataset-v2
3. `done on 4-turn augmented and real multi-turn data directly`

**Comment by siddharth1712** [comment]
Hi thank you for the information. It is very helpful.
I have trained SALM-duplex with the data following the format and datasets outlined in the paper, but I see that during inference it is not able to reply to my voice, although it is able to do well on the validation set which consists of varying speakers from the ASR data subset and other conversations datasets. Did you face this problem while training SALM-Duplex? Are there any data augmentations you do during training to increase variability and variety in the data? It would be a great help if you could answer these queries.

**Comment by zhehuaichen** [comment]
Yes data augmentation is necessary to let the model learn how to take turn. Otherwise the model may overfit to only start talking when seeing "perfect silence" from the user side. We will share more details after we wrap up the release.

**Comment by zhehuaichen** [comment]
[Nemotron 3 VoiceChat](https://build.nvidia.com/nvidia/nemotron-voicechat/modelcard) is now released in Early Access.  Try out [the demo](https://build.nvidia.com/nvidia/nemotron-voicechat) and apply for [early access](https://developer.nvidia.com/nemotron-voicechat-early-access) from NGC .
